### PR TITLE
cfgstruct.Bind: support nested config paths

### PR DESCRIPTION
**Pull request by jtolds**
this will allow some config cleanups in a future PR

### CODE REVIEW

File: pkg/cfgstruct/bind.go [hunk 1]
```diff
@@ -14,15 +15,24 @@ import (
...
+type BindOpt func(vars map[string]confVar)
 
-// ConfDir sets a variable for default options called $CONFDIR
-func ConfDir(confdir string) BindOpt {
-	return BindOpt(func(vars map[string]string) {
-		vars["CONFDIR"] = os.ExpandEnv(confdir)
+// ConfDir sets variables for default options called $CONFDIR and $CONFNAME.
+// If nested is true, appends the parent struct field name to the paths before
+// descending into substructs.
+func ConfDir(confdir string, nested bool) BindOpt {
```
[the remark this answers is not in the document]
Comment (jtolds) [reply pkg/cfgstruct/bind.go]: we use this feature in #791, but you're right that only a few test programs use unnested behavior. everything switches to nested. should i merge the two PRs?

[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/cfgstruct/bind.go]: It's better if they are separate, but if it makes your life easier, you can also merge.

[the remark this answers is not in the document]
Comment (jtolds) [reply pkg/cfgstruct/bind.go]: i agree they're better separate, but if they're separate, i can't change the default behavior of ConfDir in this change.

[the remark this answers is not in the document]
Comment (jtolds) [reply pkg/cfgstruct/bind.go]: perhaps what i do is make ConfDirNested and then in the next change rename and get rid of the old ConfDir?

[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/cfgstruct/bind.go]: Sounds good

**Comment by jtolds** [comment]
fixed

File: pkg/cfgstruct/bind.go [hunk 2]
```diff
@@ -107,7 +138,8 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
 				check(err)
 				flags.Float64Var(fieldaddr.(*float64), flagname, val, help)
 			case reflect.TypeOf(string("")):
-				flags.StringVar(fieldaddr.(*string), flagname, expand(vars, def), help)
+				flags.StringVar(
+					fieldaddr.(*string), flagname, expand(resolvedVars, def), help)
```
Comment (egonelbre) [inline pkg/cfgstruct/bind.go]: Single line

File: cmd/captplanet/setup.go [hunk 3]
```diff
@@ -46,8 +46,7 @@ var (
 func init() {
 	rootCmd.AddCommand(setupCmd)
 	cfgstruct.Bind(setupCmd.Flags(), &setupCfg,
-		cfgstruct.ConfDir(defaultConfDir),
-	)
+		cfgstruct.ConfDir(defaultConfDir))
```
Comment (egonelbre) [inline cmd/captplanet/setup.go]: Single line

**Pull request merged**